runtime/src/main/java/io/quarkiverse/resteasy/problem/validation/ConstraintViolationExceptionMapper.java

```diff
@@ -140,4 +147,24 @@ private String serializePath(Path propertyPath, int skipFirstSegments) {
         return String.join(".", pathSegments);
     }
 
+    private boolean isDeclarativeValidation(ConstraintViolation<?> violation) {
+        return hasJaxRsContext() &&
+                (rootBeanResourceClassMatches(violation) || propertyPathStartsWithMethod(violation));
```

Comment:
Nitpick: I personally find it more readable to expand complex if conditions into multiple ifs like this:
```suggestion
        if(!hasJaxRsContext()) {
            return false;
        }
        if(rootBeanResourceClassMatches(violation)) {
            return true;
        }
        if(propertyPathStartsWithMethod(violation))) {
            return true;
        }
        return false;
```

But feel free to ignore this suggestion.